### ISSUE TITLE
Can't use ref on Switch component

### ISSUE THREAD

**Issue by herrethan** [system-ui/theme-ui#2171]
**Describe the bug**
Can't use ref on Switch component when creating an HOC. 

**To Reproduce**

```js
export const CustomSwitch = React.forwardRef<HTMLInputElement, SwitchProps>((props, ref) => {
  return <Switch ref={ref} {...props}  />;
}
```
Throws the TS error 
```
Type 'string' is not assignable to type 'Ref<HTMLInputElement> | undefined'
```

There seems to be a clash between `props.ref` and the forwardedRef, where `props.ref` uses a LegacyRef which accepts a string.

**Hint**
I believe we just need to update `ComponentProps` to `ComponentPropsWithRef` here?

[  extends Assign<React.ComponentProps<'input'>, BoxOwnProps> {](https://github.com/system-ui/theme-ui/blob/6f09ac088e98531080da7b282ea6311a6c0bdd44/packages/components/src/Switch.tsx#L14)

**Comment by herrethan** [comment]
Note: workaround I'm using right now is to just remove the offending `ref` from the props
```ts
React.forwardRef<HTMLInputElement, SwitchProps>(({ ref: legacyRefNotToBeUsed, ...props}, ref) =>
```

**Comment by DParfitt** [comment]
I think this is now the case with all components. I just upgraded to 0.14 and now no refs can be passed down to `Box` and `Flex` either.

**Comment by hasparus** [comment]
Thanks for the issue @herrethan!

**Comment by hasparus** [comment]
Hey folks, I just pushed a fix for this. Releasing after CI passes.

Thank you for updating to 0.14 so quickly and finding this!

**Comment by hasparus** [comment]
Should be fixed in [v0.14.1](https://github.com/system-ui/theme-ui/releases/tag/v0.14.1). Feel free to reopen if you have still experience this issue.

**Comment by herrethan** [comment]
@hasparus thanks for the quick turnaround!
